File: Easy/registry/meta_registryamministrativiview/meta_registryamministrativiview.cs

/*
Easy
Copyright (C) 2021 Università degli Studi di Catania (www.unict.it)
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/


using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_registryamministrativiview
{
    public class Meta_registryamministrativiview : Meta_easydata 
	{
        public Meta_registryamministrativiview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "registryamministrativiview") {
				Name = "Personale Amministrativo";
			EditTypes.Add("amministrativi");
			ListingTypes.Add("amministrativi");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idreg"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "amministrativi": {
						DescribeAColumn(T, "title_description", "Titolo", nPos++);
						DescribeAColumn(T, "registry_surname", "Cognome", nPos++);
						DescribeAColumn(T, "registry_forename", "Nome", nPos++);
						DescribeAColumn(T, "registry_cf", "Codice fiscale", nPos++);
						DescribeAColumn(T, "registry_badgecode", "Codice bedge", nPos++);
						DescribeAColumn(T, "registry_active", "Attivo", nPos++);
						DescribeAColumn(T, "contrattokind_title", "Tipo", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/registry/meta_registryamministrativiview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_registryamministrativiview() {
        MetaData.apply(this, ["registryamministrativiview"]);
        this.name = 'meta_registryamministrativiview';
    }

    meta_registryamministrativiview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_registryamministrativiview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'amministrativi':
						this.describeAColumn(table, 'title_description', 'Titolo', null, 1200, 50);
						this.describeAColumn(table, 'registry_surname', 'Cognome', null, 2000, 50);
						this.describeAColumn(table, 'registry_forename', 'Nome', null, 3000, 50);
						this.describeAColumn(table, 'registry_cf', 'Codice fiscale', null, 4000, 16);
						this.describeAColumn(table, 'registry_badgecode', 'Codice badge', null, 5000, 20);
						this.describeAColumn(table, 'registry_active', 'Attivo', null, 10000, null);
						this.describeAColumn(table, 'XXregistrylegalstatus', 'Servizi di ruolo - Contratti', null, 52000, null);
//$objCalcFieldConfig_amministrativi$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idreg"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "amministrativi": {
						return "title asc ";
					}
					case "amministrativi": {
						return "registry_surname asc , registry_forename asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('registryamministrativiview', new meta_registryamministrativiview('registryamministrativiview'));

	}());
